decompyle3/semantics/customize38.py

```diff
@@ -303,41 +303,53 @@ def n_formatted_value_debug(node):
         p = self.prec
         self.prec = 100
 
-        formatted_value = node[1]
-        value_equal = node[0].attr
-        assert formatted_value.kind.startswith("formatted_value")
         old_in_format_string = self.in_format_string
-        self.in_format_string = formatted_value.kind
-        format_value_attr = node[-1]
+        self.in_format_string = "formatted_value_debug"
 
-        post_str = ""
-        if node[-1] == "BUILD_STRING_3":
-            post_load_str = node[-2]
-            post_str = self.traverse(post_load_str, indent="")
-            post_str = strip_quotes(post_str)
-
-        if format_value_attr == "FORMAT_VALUE_ATTR":
-            attr = format_value_attr.attr
-            if attr & 4:
-                fmt = strip_quotes(self.traverse(node[3], indent=""))
-                attr_flags = attr & 3
-                if attr_flags:
-                    conversion = "%s:%s" % (
-                        FSTRING_CONVERSION_MAP.get(attr_flags, ""),
-                        fmt,
-                    )
+        # Collect all expressions that we're formatting one by one
+        # Fragments is a list of parsed expressions, like ['a', '{a=}', 'b']
+        # Caveat: python merges strings on bytecode level, so for example
+        # f"user setting: {user=}" becomes f"settings: user={user!r}".
```

Comment:
A small typo which is my mistake `f"settings: ` -> `f"setting: `

Reply:
Right. Missed that. Fixed.